modules/nodes-pods-autoscaling-custom-trigger.adoc

```diff
@@ -0,0 +1,157 @@
+// Module included in the following assemblies:
+//
+// * nodes/nodes-pods-autoscaling-custom.adoc
+
+:_content-type: CONCEPT
+[id="nodes-pods-autoscaling-custom-trigger_{context}"]
+= Understanding the custom metrics autoscaler triggers
+
+A trigger, or scaler, provide the metrics that the Horizontal Pod Autoscaler uses to scale your pods. 
+
+The following are examples of some triggers you can use with the custom metrics autoscaler (CMA). For a full list of available triggers, see the link:https://keda.sh/docs/2.7/scalers[Keda Scalers] documentation. Because the custom metrics autoscaler is Technology Preview, Red Hat does not recommend using the triggers/scalers in production. 
+
+You use a `ScaledObject` or `ScaledJob` to configure triggers for specific objects, as described in the sections that follow. 
+
+[discrete]
+[id="nodes-pods-autoscaling-custom-trigger-prom_{context}"]
+== Understanding the Prometheus trigger
```

Comment:
We should add a paragraph or subsection here, in case users would like to scale apps based on Prometheus, that is part of OpenShift Monitoring -> specific setup is required here. I described it here: https://github.com/zroubalik/keda-openshift-examples/tree/main/prometheus/ocp-monitoring

Reply:
@zroubalik 
Can people use the Monitoring docs here?
https://docs.openshift.com/container-platform/4.10/monitoring/configuring-the-monitoring-stack.html#enabling-query-logging-for-thanos-querier_configuring-the-monitoring-stack

Reply:
Yeah, this is a great reference for enabling of monitoring for user workloads (this is an prerequisite).

But then they need to:
[1. create a ServiceAccount to get a token](https://github.com/zroubalik/keda-openshift-examples/tree/main/prometheus/ocp-monitoring#2-create-a-service-account-or-reuse-existing-one-locate-assigned-token) 
[2. create a Role](https://github.com/zroubalik/keda-openshift-examples/tree/main/prometheus/ocp-monitoring#4-create-a-role-for-reading-metric-from-thanos) 
[3. add that role to the previously created Service Account](https://github.com/zroubalik/keda-openshift-examples/tree/main/prometheus/ocp-monitoring#5-add-the-role-for-reading-metrics-from-thanos-to-the-service-account).
[4. and finally reference the token (secret) in the TriggerAuthentication object used by Prometheus Scaler](https://github.com/zroubalik/keda-openshift-examples/tree/main/prometheus/ocp-monitoring#3-define-triggerauthentication-with-the-service-accounts-token)

We should cover this in the docs, becuase it is not very straightforward and we should enable users to use OpenShift Monitoring Prometheus (Thanos) instance.

Reply:
A section like `How to scale based on metrics from OpenShift Monitoring`

Reply:
See: [Configuring the custom metrics autoscaler to use Prometheus Cluster Monitoring](http://file.rdu.redhat.com/~mburke/OSDOCS-3048-custom-autoscaling-keda/nodes/pods/nodes-pods-autoscaling-custom.html#nodes-pods-autoscaling-custom-prometheus-config_nodes-pods-autoscaling-custom)

Reply:
looking good, could you please add a link to that section in here? So users reading about Prometheus trigger could quickly check this guide?